snippets/csharp/System.Xml/XmlDocument/Overview/xmlhelpermethods.cs

```diff
@@ -91,45 +90,40 @@ private string generateXMLString()
         //************************************************************************************
         public XmlDocument LoadDocumentWithSchemaValidation(bool generateXML, bool generateSchema)
         {
-            XmlReader reader;
-
+            XmlReader reader = null;
             XmlReaderSettings settings = new XmlReaderSettings();
-
             // Helper method to retrieve schema.
             XmlSchema schema = getSchema(generateSchema);
 
-            if (schema == null)
-            {
-                return null;
-            }
-
             settings.Schemas.Add(schema);
-
             settings.ValidationEventHandler += ValidationCallback;
             settings.ValidationFlags =
                 settings.ValidationFlags | XmlSchemaValidationFlags.ReportValidationWarnings;
             settings.ValidationType = ValidationType.Schema;
-
-            try
-            {
-                reader = XmlReader.Create("booksData.xml", settings);
-            }
-            catch (System.IO.FileNotFoundException)
+            if (!generateXML)
             {
-                if (generateXML)
+                try
                 {
-                    string xml = generateXMLString();
-                    byte[] byteArray = Encoding.UTF8.GetBytes(xml);
-                    MemoryStream stream = new MemoryStream(byteArray);
-                    reader = XmlReader.Create(stream, settings);
+                    reader = XmlReader.Create("booksData.xml", settings);
                 }
-                else
+                catch (FileNotFoundException ex)
                 {
-                    return null;
+                    Console.WriteLine(
+                        $"XML file not found so generating: {ex.Message}");
```

Comment:
can you check `File.Exists` and let the exception happen instead?

Reply:
I am not sure what you are asking. If I check File.Exists then I can avoid the FileNotFoundException all together. I checked File.Exists in my first attempt to clean up and avoided FileNotFoundException . I was told by a reviewer that the example is to show the XmlReader.Create throws a documented exception of FIleNotFoundException. I should handle the documented exception gracefully.  The key to the changes 1) Removing .NET 1.0 code for handling Memory Stream (moved to .NET 1.1 which is cleaner) and 2) the generateXML parameter is always used in the method. In the original example the generateXML parameter was ignored depending on if errors manifested themselves. Setting generateXML=true was simply ignored in the original code at times.